File: number_theory/euclidean_algorithm/cxx/gcd_iterate.cpp
#include <cassert>
#include <utility>
#include <iostream>

long Euclid(long a, long b) {
    assert(a >= b && b >= 0);
    auto r = a % b;
    while (r != 0) {
        a = b;
        b = r;
        r = a % b;
    }
    return b;
}

long gcd(long a, long b) {
    assert(a >= 0 && b >= 0);
    if (a < b) {
        std::swap(a, b);
    }
    return Euclid(a, b);
}

void print_gcd(long a, long b) {
    std::cout << "gcd(" << a << ", " << b << ") = " << gcd(a, b) << std::endl;
}

int main() {
    print_gcd(252, 198);
}

File: number_theory/extended_euclidean_algorithm/python/recipe-02/gcd_iterate.py

def Euclid(a:int, b:int):
    assert a >= b >= 0
    r = a % b
    while r != 0:
        a = b
        b = r
        r = a % b
    return b

def gcd(a:int, b:int):
    assert a >= 0 and b >= 0
    if a < b:
        a, b = b, a # swap(a,b)
    return Euclid(a, b)

def print_gcd(a, b):
    print("gcd({}, {}) = {}".format(a, b, gcd(a, b)))

if __name__ == "__main__":
    print_gcd(252, 198)
